rlclientlib/logger/http_transport_client.h

```diff
@@ -90,4 +89,7 @@ namespace reinforcement_learning {
     i_trace* _trace;
     error_callback_fn* _error_callback;
   };
+
+  template class http_transport_client<apim_http_authorization>;
```

Comment:
This does not seem necessary here.  Any reason for this?

Reply:
![image](https://user-images.githubusercontent.com/14878919/133306579-39f1175e-b26d-4b18-b5c2-a917d87f12e0.png)

Reply:
If I remove these definitions and build rl_sim_cpp I get linking errors. Google search suggested this. Not sure if there is a better way. rlclientlib builds fine. @rajan-chari

Reply:
1>Done building project "rlclientlib.vcxproj".
2>------ Build started: Project: rl_sim_cpp, Configuration: Debug x64 ------
2>rlclientlib.lib(azure_factories.obj) : error LNK2019: unresolved external symbol "public: __cdecl reinforcement_learning::http_transport_client<class reinforcement_learning::apim_http_authorization>::http_transport_client<class reinforcement_learning::apim_http_authorization>(class reinforcement_learning::i_http_client *,unsigned __int64,unsigned __int64,class reinforcement_learning::i_trace *,class reinforcement_learning::error_callback_fn *)" (??0?$http_transport_client@Vapim_http_authorization@reinforcement_learning@@@reinforcement_learning@@QEAA@PEAVi_http_client@1@_K1PEAVi_trace@1@PEAVerror_callback_fn@1@@Z) referenced in function "int __cdecl reinforcement_learning::create_apim_http_api_sender(class reinforcement_learning::i_sender * *,class reinforcement_learning::utility::configuration const &,char const *,int,int,class reinforcement_learning::error_callback_fn *,class reinforcement_learning::i_trace *,class reinforcement_learning::api_status *)" (?create_apim_http_api_sender@reinforcement_learning@@YAHPEAPEAVi_sender@1@AEBVconfiguration@utility@1@PEBDHHPEAVerror_callback_fn@1@PEAVi_trace@1@PEAVapi_status@1@@Z)
2>rlclientlib.lib(azure_factories.obj) : error LNK2019: unresolved external symbol "public: __cdecl reinforcement_learning::http_transport_client<class reinforcement_learning::eventhub_http_authorization>::http_transport_client<class reinforcement_learning::eventhub_http_authorization>(class reinforcement_learning::i_http_client *,unsigned __int64,unsigned __int64,class reinforcement_learning::i_trace *,class reinforcement_learning::error_callback_fn *)" (??0?$http_transport_client@Veventhub_http_authorization@reinforcement_learning@@@reinforcement_learning@@QEAA@PEAVi_http_client@1@_K1PEAVi_trace@1@PEAVerror_callback_fn@1@@Z) referenced in function "int __cdecl reinforcement_learning::interaction_sender_create(class reinforcement_learning::i_sender * *,class reinforcement_learning::utility::configuration const &,class reinforcement_learning::error_callback_fn *,class reinforcement_learning::i_trace *,class reinforcement_learning::api_status *)" (?interaction_sender_create@reinforcement_learning@@YAHPEAPEAVi_sender@1@AEBVconfiguration@utility@1@PEAVerror_callback_fn@1@PEAVi_trace@1@PEAVapi_status@1@@Z)
2>E:\Repo\upstream_rl\x64\Debug\rl_sim_cpp.exe : fatal error LNK1120: 2 unresolved externals
2>Done building project "rl_sim_cpp.vcxproj" -- FAILED.

Reply:
I believe we need to move http_transport_client.cc to http_transport_client.h in order to fix that.
It will mess the diff though. Maybe we can leave these lines for now and create another refactoring PR (http_transport_client.h -> http_transport_client.cc + instantiation removal) immediately after that.
@slahabar , @rajan-chari , what do you think?

Reply:
Sounds good @ataymano , @slahabar